src/core/document.js

```diff
@@ -62,138 +61,211 @@ var Page = (function PageClosure() {
+      return shadow(this, 'cropBox', this.mediaBox);
+    }
+    return shadow(this, 'cropBox', cropBox);
+  }
 
-    get userUnit() {
-      var obj = this.pageDict.get('UserUnit');
-      if (!isNum(obj) || obj <= 0) {
-        obj = DEFAULT_USER_UNIT;
-      }
-      return shadow(this, 'userUnit', obj);
-    },
+  get userUnit() {
+    let obj = this.pageDict.get('UserUnit');
+    if (!isNum(obj) || obj <= 0) {
+      obj = DEFAULT_USER_UNIT;
+    }
+    return shadow(this, 'userUnit', obj);
+  }
 
-    get view() {
-      // From the spec, 6th ed., p.963:
-      // "The crop, bleed, trim, and art boxes should not ordinarily
-      // extend beyond the boundaries of the media box. If they do, they are
-      // effectively reduced to their intersection with the media box."
-      var mediaBox = this.mediaBox, cropBox = this.cropBox;
-      if (mediaBox === cropBox) {
-        return shadow(this, 'view', mediaBox);
-      }
-      var intersection = Util.intersect(cropBox, mediaBox);
-      return shadow(this, 'view', intersection || mediaBox);
-    },
+  get view() {
+    // From the spec, 6th ed., p.963:
+    // "The crop, bleed, trim, and art boxes should not ordinarily
+    // extend beyond the boundaries of the media box. If they do, they are
+    // effectively reduced to their intersection with the media box."
+    const mediaBox = this.mediaBox, cropBox = this.cropBox;
+    if (mediaBox === cropBox) {
+      return shadow(this, 'view', mediaBox);
+    }
+
+    const intersection = Util.intersect(cropBox, mediaBox);
+    return shadow(this, 'view', intersection || mediaBox);
+  }
+
+  get rotate() {
+    let rotate = this._getInheritableProperty('Rotate') || 0;
+
+    // Normalize rotation so it's a multiple of 90 and between 0 and 270.
+    if (rotate % 90 !== 0) {
+      rotate = 0;
+    } else if (rotate >= 360) {
+      rotate = rotate % 360;
+    } else if (rotate < 0) {
+      // The spec doesn't cover negatives. Assume it's counterclockwise
+      // rotation. The following is the other implementation of modulo.
+      rotate = ((rotate % 360) + 360) % 360;
+    }
+    return shadow(this, 'rotate', rotate);
+  }
 
-    get rotate() {
-      var rotate = this._getInheritableProperty('Rotate') || 0;
-      // Normalize rotation so it's a multiple of 90 and between 0 and 270
-      if (rotate % 90 !== 0) {
-        rotate = 0;
-      } else if (rotate >= 360) {
-        rotate = rotate % 360;
-      } else if (rotate < 0) {
-        // The spec doesn't cover negatives, assume its counterclockwise
-        // rotation. The following is the other implementation of modulo.
-        rotate = ((rotate % 360) + 360) % 360;
+  getContentStream() {
+    const content = this.content;
+    let stream;
+
+    if (Array.isArray(content)) {
+      // Fetching the individual streams from the array.
+      const xref = this.xref;
+      const streams = [];
+      for (let i = 0, ii = content.length; i < ii; ++i) {
```

Comment:
Optional nit: Could probably use `for ... of` instead here.

Reply:
Somehow I keep forgetting about this. Thank you!